appserver/persistence/cmp/ejb-mapping/src/main/java/com/sun/jdo/api/persistence/mapping/ejb/SunCmpMappingsUtils.java

```diff
@@ -176,24 +176,23 @@ else if (l.size() > 1)
      * @param propVal the value of the element
      * @param type The expected type of the value to be returned.
      * @throws IllegalArgumentException If the bean is not part of a complete bean graph.
-     */    
-    protected static List findCompatibleBeansWithValue(BaseBean root, String propName, String propVal, Class type) throws IllegalArgumentException {
-        List retVal = null; 
+     */
+    protected static List<?> findCompatibleBeansWithValue(BaseBean root, String propName, String propVal, Class<?> type) throws IllegalArgumentException {
         GraphManager gm = root.graphManager();
-        if (null == gm) 
-            throw new IllegalArgumentException(
-                    bundle.getString("ERR_DISCONNECTED_NOT_SUPPORTED"));
+        if (null == gm) {
+            throw new IllegalArgumentException(bundle.getString("ERR_DISCONNECTED_NOT_SUPPORTED"));
+        }
         String[] props = root.findPropertyValue(propName, propVal);
-        int len = 0;
-        if (null != props)
-            len = props.length;
-        if (len > 0)
-            retVal = new ArrayList(); 
-        for (int i = 0; i < len; i++) {
+        if (props == null) {
+            return null;
+        }
+        List<BaseBean> retVal = new ArrayList<>(); 
+        for (int i = 0; i < props.length; i++) {
             // get the bean that is the property's parent.
-            BaseBean candidate = gm.getPropertyParent(props[i]); 
```

Comment:
sonar can't do index based exclusion so I rewrote the body to both make it more readable for humans and tools.